File: ExecutionContext/MultipleOrderedEC-C++/AddButton2.cpp
/*!
* @file  AddButton2.cpp
* @brief 並列ブロック追加ボタン、並列ブロックの先頭に直列ブロック追加ボタン
*
*/

#include "AddButton2.h"

#include <QtWidgets>



/**
*@brief 並列ブロック追加ボタン、並列ブロックの先頭に直列ブロック追加ボタンのコンストラクタ
* @param text 表示テキスト
* @param parent 親ウィジェット
*/
AddButton2::AddButton2(QString text, QWidget *parent)
    : QWidget(parent)
{
	PB = new QPushButton(text);

	QVBoxLayout *mainLayout = new QVBoxLayout;


	connect(PB, SIGNAL(clicked()),
            this, SLOT(clickedSlot()));

	mainLayout->addWidget(PB);
	



	setLayout(mainLayout);
}

/**
*@brief ボタンクリック時に呼び出すスロット
*/
void AddButton2::clickedSlot()
{
	clicked(Cl);
}

File: ExecutionContext/MultipleOrderedEC-Python/MultipleOrderedEC_Lib/AddButton2.py
# -*- coding: utf-8 -*-
##
#   @file AddButton2.py
#   @brief 並列ブロック追加ボタン、並列ブロックの先頭に直列ブロック追加ボタン


from PyQt4 import QtCore, QtGui
from ExComp import CompLayout



##
# @class AddButton2
# @brief 直列ブロック追加ボタンを含むウィジェット
#
class AddButton2(QtGui.QWidget):
    clicked = QtCore.pyqtSignal(object)
    ##
    # @brief コンストラクタ
    # @param self 
    # @param text 表示テキスト
    # @param parent 親ウィジェット
    def __init__(self, text, parent=None):
        super(AddButton2, self).__init__(parent)
        
        self.Cl = None
        self.PB = QtGui.QPushButton(text)
        self.mainLayout = QtGui.QVBoxLayout()
        
        #connect(PB, SIGNAL(clicked()),
        #    this, SLOT(clickedSlot()))

        self.PB.clicked.connect(self.clickedSlot)

        self.mainLayout.addWidget(self.PB)

        self.setLayout(self.mainLayout)

    
    ##
    # @brief ボタンクリック時に呼び出すスロット
    # @param self 
    def clickedSlot(self):
        
        self.clicked.emit(self.Cl)
